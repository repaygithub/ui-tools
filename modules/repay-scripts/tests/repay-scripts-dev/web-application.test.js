const TestSetup = require('../test-setup')

describe('@repay/repay-scripts', () => {
  it('should build a web application for local development', async () => {
    const test = new TestSetup('web-application')
    await test.setup()
    test.spawn('yarn', [
      'repay-scripts',
      'dev',
      '--babel-env',
      'production',
      'src/index.js',
      '--port',
      '9696',
    ])

    await test.waitForText('Compiled successfully')
    const page = await test.getPage()
    await page.goto('https://localhost:9696')
    const content = await page.$eval('.app-root', (el) => el.outerHTML)
    expect(content).toMatchSnapshot()
  })

  it('should build a web application for local development using a custom index.html passed from CLI', async () => {
    const test = new TestSetup('web-application')
    await test.setup()
    test.spawn('yarn', [
      'repay-scripts',
      'dev',
      '--babel-env',
      'production',
      'src/index.js',
      '--port',
      '9696',
      '-t',
      'template.html',
    ])
    await test.waitForText('Compiled successfully')
    const page = await test.getPage()
    await page.goto('https://localhost:9696')
    const content = await page.$eval("head > meta[name='viewport']", (el) => el.content)
    expect(content).toBe('custom-location')
  })

  it('should build a web application for local development using a custom index.html found in the default location', async () => {
    const test = new TestSetup('web-application')
    await test.setup()
    test.spawn('yarn', [
      'repay-scripts',
      'dev',
      '--babel-env',
      'production',
      'src/index.js',
      '--port',
      '9696',
    ])
    await test.waitForText('Compiled successfully')
    const page = await test.getPage()
    await page.goto('https://localhost:9696')
    const content = await page.$eval("head > meta[name='viewport']", (el) => el.content)
    expect(content).toBe('test')
  })
})
