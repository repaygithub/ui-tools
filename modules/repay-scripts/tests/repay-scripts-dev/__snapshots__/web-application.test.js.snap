// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@repay/repay-scripts should build a web application for local development 1`] = `"<div class=\\"app-root\\">This is the application root</div>"`;
